src/npm.ts

```diff
@@ -0,0 +1,50 @@
+import { Job } from "@brigadecore/brigadier";
+import { Project } from "@brigadecore/brigadier/out/events";
+
+const nodeImage = "node:12.3.1-stretch"
+const defaultReleaseTagRegex = /^refs\/tags\/v([0-9]+(?:\.[0-9]+)*(?:\-.+)?)$/;
+
+export class NPM extends Job {
+
+    public releaseTagRegex: RegExp;
+    public buildJob: Job;
+    public version: string;
+    public project: Project;
+
+    constructor(name: string, build: Job, version: string, project: Project, image?: string, releaseRegex?: RegExp) {
+        super(name, image);
+
+        this.buildJob = build;
+        this.version = version;
+        this.project = project;
+
+        if (image != undefined) {
+            this.image = image;
+        } else {
+            this.image = nodeImage;
+        }
+
+        if (releaseRegex != undefined) {
+            this.releaseTagRegex = releaseRegex;
+        } else {
+            this.releaseTagRegex = defaultReleaseTagRegex;
+        }
+
+        this.env = {
+            "NPM_TOKEN": this.project.secrets.npmToken
+        };
+
+        // The steps to publish include the steps to build, and then some, so we'll
+        // use the build job steps as a starting point.
+        this.tasks = this.buildJob.tasks;
+
+        // If we leave .npmrc at the root of the project with the NPM_TOKEN env var
+        // unset, all yarn commands will fail. Since this env var is populated with
+        // the correct secret ONLY for this one job, we create .npmrc right here,
+        // just in time.
+        this.tasks.push("echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}' > .npmrc");
+        this.tasks.push("sed -i 's/^  \"version\":.*/  \"version\": \"" + version + "\",/' package.json");
```

Comment:
I realize this `sed` command is copied verbatim from https://github.com/brigadecore/brigadier/pull/28, but this won't work with everyone's `package.json` since it's sensitive to how the JSON is formatted. There really isn't any way to make this foolproof using `sed`.

The right way to do this would be with a tool like `jq`, but that's not a tool we can safely assume is available. Indeed, it's not available on the default `node:12.3.1-stretch` image.

We could add a step to install jq, but then we need to know something about which package manager the image's OS uses.

I'm going to go back and fix https://github.com/brigadecore/brigadier/pull/28 since there we have the luxury of knowing what package manager is in use and even there, jq will be safer and cleaner than sed. But right here, I still don't know how to generalize this.

Reply:
@radu-matei @krancour shall we just go with the `jq` approach used in https://github.com/brigadecore/brigadier/pull/28 ?

I'd love to move this through, with the benefit of release automation in this repo and of course re-usability for other repos.

@radu-matei I can contribute to this branch if you'll be busy; let me know.  We can test the next brigade-utils release once we think we've got it.